relay-server/src/actors/processor.rs

```diff
@@ -1274,6 +1282,77 @@ impl EnvelopeProcessorService {
         }
     }
 
+    fn handle_submit_envelope(&self, message: SubmitEnvelope) {
+        let SubmitEnvelope { mut envelope } = message;
+
+        #[cfg(feature = "processing")]
+        if self.inner.config.processing_enabled() {
+            if let Some(store_forwarder) = self.inner.store_forwarder.clone() {
+                relay_log::trace!("sending envelope to kafka");
+                store_forwarder.send(StoreEnvelope { envelope });
+                return;
+            }
+        }
+
+        // if we are in capture mode, we stash away the event instead of forwarding it.
+        if Capture::should_capture(&self.inner.config) {
+            relay_log::trace!("capturing envelope in memory");
+            self.inner
+                .test_store
+                .send(Capture::accepted(envelope.take_envelope()));
+            envelope.accept();
+            return;
+        }
+
+        relay_log::trace!("sending envelope to sentry endpoint");
+        let http_encoding = self.inner.config.http_encoding();
+        match encode_envelope(envelope.envelope_mut(), http_encoding) {
```

Comment:
Note that setting the sent_at timestamp (part of the body), serialization, and HTTP encoding are now a single fallible step. In a follow up, we can check if sent_at could be lifted out into a header.